app/org/sagebionetworks/bridge/models/accounts/SignUp.java

```diff
@@ -1,42 +1,32 @@
 package org.sagebionetworks.bridge.models.accounts;
 
+import java.util.Objects;
 import java.util.Set;
 
 import org.sagebionetworks.bridge.Roles;
-import org.sagebionetworks.bridge.json.JsonUtils;
 import org.sagebionetworks.bridge.models.BridgeEntity;
 
-import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.collect.Sets;
 
-public class SignUp implements BridgeEntity {
-
-    private static final String EMAIL_FIELD = "email";
-    private static final String USERNAME_FIELD = "username";
-    private static final String PASSWORD_FIELD = "password";
-    private static final String ROLES_FIELD = "roles";
+public final class SignUp implements BridgeEntity {
 
     private final String username;
     private final String email;
     private final String password;
     private final Set<Roles> roles;
+    private final Set<String> dataGroups;
 
-    public SignUp(String username, String email, String password, Set<Roles> roles) {
+    @JsonCreator
+    public SignUp(@JsonProperty("username") String username, @JsonProperty("email") String email, 
+            @JsonProperty("password") String password, @JsonProperty("roles") Set<Roles> roles, 
+            @JsonProperty("dataGroups") Set<String> dataGroups) {
         this.username = username;
         this.email = email;
         this.password = password;
-        this.roles = Sets.newHashSet();
-        if (roles != null) {
-            this.roles.addAll(roles);
-        }
-    }
-
-    public static final SignUp fromJson(JsonNode node, boolean allowRoles) {
-        String username = JsonUtils.asText(node, USERNAME_FIELD);
-        String email = JsonUtils.asText(node, EMAIL_FIELD);
-        String password = JsonUtils.asText(node, PASSWORD_FIELD);
-        Set<Roles> roles = (allowRoles) ? JsonUtils.asRolesSet(node, ROLES_FIELD) : null;
-        return new SignUp(username, email, password, roles);
+        this.roles = (roles == null) ? Sets.newHashSet() : roles;
+        this.dataGroups = (dataGroups == null) ? Sets.newHashSet() : dataGroups;
```

Comment:
Added. I'm not adding tests because we don't rely on this behavior in these cases, it's more like a runtime check against programming errors.